src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -98,7 +102,25 @@ public int getHttpStatus() {
   }
 
   public boolean isSuccess() {
-    return successCodes.contains(getHttpStatus());
+    boolean isSuccess = false;
+    if (checkRetryable()) {
+      isSuccess = false;
```

Comment:
why do we need need this if (checkRetryable()) check?  Can't we just return true if successCodes.contains(getHttpStatus()) - i.e. what this method was doing previously

Reply:
Previously this method: public boolean isSuccess() was returning true if the HTTP response code was 200 OK.

This included the scenario where the ServiceNow Table Rest API returned a 200 OK response for API calls taking more execution time than the specified limit (60 sec). Here, we get an error message saying Error - com.glide.sys.TransactionCancelledException: Transaction cancelled: maximum execution time exceeded in the response JSON for all REST API calls that take more than 60 sec and eventually lose that batch. [Data Loss Issue]

To overcome this, we added the retry functionality. Whenever this error is encountered, the connector will retry the API. The checkRetryable() method contains the logic as to when the retry functionality should be invoked.